packages/edit-site/src/components/page-patterns/style.scss

```diff
@@ -102,6 +106,29 @@
 	.edit-site-patterns__pattern-lock-icon {
 		fill: currentcolor;
 	}
+
+	.dataviews-view-grid & {
+		display: -webkit-box;
+		-webkit-line-clamp: 2;
+		-webkit-box-orient: vertical;
+		white-space: normal;
+		padding-top: 3px;
+		align-self: start;
+
+		button.components-button.is-link {
+			-webkit-line-clamp: 2;
+			-webkit-box-orient: vertical;
+			display: -webkit-box;
+			white-space: normal;
+		}
+	}
+}
+
+.components-button.dataviews-all-actions-button.has-icon:not(.has-text) {
+	.dataviews-view-grid & {
+		min-width: 24px;
+		height: 24px;
+	}
```

Comment:
It is better to avoid using selectors with component prefixes whenever possible. Additionally, the focus outline is not square:

![image](https://github.com/user-attachments/assets/5b5ede0a-f993-4cf5-8fde-2e759fbe0ec5)

Perhaps it would be better to add a prop to [the `CompactItemActions` component](https://github.com/WordPress/gutenberg/blob/4836360a916f18775f31383dfd2ff698e7a6c68b/packages/dataviews/src/components/dataviews-item-actions/index.tsx#L251) to change the size only when in grid layout. I'll consider addressing this in a separate PR.